### PR TITLE
Reduce test time

### PR DESCRIPTION
**Pull request by nbodini**
@jordanperr , I have reduced the time taken by the tests to run, but it's not as low as you were hoping. From a "scientific" point of view, I honestly don't know how to further reduce it (the number of iterations is already super low, and I have removed a couple of the setups that were being tested before). Happy to further work on this if you have better ideas on how to do this.

### CODE REVIEW

**Comment by jordanperr** [comment]
Thanks, Nicola. How would you feel about splitting the tests into another section. Maybe have "scientific" tests and "code" test modules. Github would automatically run the "code" tests for all pull requests, but it would be up to developers to run the "scientific" tests on their local machine. We could, potentially, have the "scientific" tests run in an automated fashion in the repository a couple times per week.

**Comment by jordanperr** [comment]
@nbodini - In light of the decision to move to Unit tests only on pull requests, would you still like me to merge in these changes? Developers are still expected to run the full test suite locally before making pull requests, so I think there is still benefit to keeping the tests as quick as possible.

**Comment by nbodini** [comment]
@jordanperr yes, that makes sense to me (as long as we acknowledge that the reduced number of Monte Carlo iterations cannot really be considered a 100% rigorous test of the science behind uncertainty quantification).

**Comment by jordanperr** [comment]
@nbodini, would you please enable "allow edits from maintainers" on this pull request? I resolved the conflict but can not push my changes to your branch. Thank you.

https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

File: test/regression/int_pruf_plant_analysis.py [hunk 1]
```diff
@@ -1,297 +1,293 @@
...
+            s.curt_pct.std() / s.curt_pct.mean() * 100,
+        ]
 
-        nptest.assert_array_almost_equal(expected_results, calculated_results, decimal=-1)
+        nptest.assert_array_almost_equal(expected_results, calculated_results)
 
     def check_simulation_results_etr_daily(self, s):
         # Make sure AEP results are consistent to one decimal place
-        expected_results = [13.83, 15.83, 1.35, 4.29, 0.09, 4.29]
+        expected_results = [12.938536, 8.561798, 1.334704, 5.336189, 0.057874, 11.339976]
```
Comment (jordanperr) [inline test/regression/int_pruf_plant_analysis.py:275]: These have changed a lot. Are these changes expected?

File: test/regression/int_pruf_plant_analysis.py [hunk 2]
```diff
@@ -1,297 +1,293 @@
...
-                              s.curt_pct.mean() * 100,
-                              s.curt_pct.std() / s.curt_pct.mean() * 100, ]
+        nptest.assert_array_almost_equal(expected_results, calculated_results)
 
-        nptest.assert_array_almost_equal(expected_results, calculated_results, decimal=-1)
-        
     def check_simulation_results_gbm_daily(self, s):
         # Make sure AEP results are consistent to one decimal place
-        expected_results = [12.82, 14.87, 1.35, 5.17, 0.09, 5.17]
+        expected_results = [12.794527, 10.609839, 1.298789, 4.849577, 0.050383, 8.620032]
```
Comment (jordanperr) [inline test/regression/int_pruf_plant_analysis.py:260]: These have changed a lot. Are these changes expected?

File: test/regression/int_pruf_plant_analysis.py [hunk 3]
```diff
@@ -1,297 +1,293 @@
...
-        expected_merra2 = pd.Series([11.02, 7.04, 8.42])
-        expected_era5 = pd.Series([10.48, 7.71, 9.61])
+        nptest.assert_array_almost_equal(expected_avail_gwh, df.loc[date_ind, "availability_gwh"])
+        nptest.assert_array_almost_equal(expected_curt_gwh, df.loc[date_ind, "curtailment_gwh"])
+        nptest.assert_array_almost_equal(expected_avail_pct, df.loc[date_ind, "availability_pct"])
+        nptest.assert_array_almost_equal(expected_curt_pct, df.loc[date_ind, "curtailment_pct"])
 
-        date_ind = pd.to_datetime(['2014-01-02', '2014-10-12', '2015-12-28'])
+    # Commenting this out because it's never called.
+    # def check_process_reanalysis_data_daily(self, df):
```
Comment (jordanperr) [inline test/regression/int_pruf_plant_analysis.py:194]: This function is never called. From which test should it be called?

File: test/regression/int_pruf_plant_analysis.py (hunk 1 above)
Comment (jordanperr) [reply test/regression/int_pruf_plant_analysis.py:275]: Confirmed these changes are reasonable in meeting.

File: test/regression/int_pruf_plant_analysis.py (hunk 2 above)
Comment (jordanperr) [reply test/regression/int_pruf_plant_analysis.py:260]: Confirmed as reasonable in meeting.

File: test/regression/int_pruf_plant_analysis.py (hunk 3 above)
Comment (jordanperr) [reply test/regression/int_pruf_plant_analysis.py]: Fixed in commit f12fc10

**Pull request merged**